src/praisonai/praisonai/scheduler/run_policy.py

```
@@ -0,0 +1,166 @@
+        applies the built-in heuristics.  Returns ``ok=True`` when scanning is
+        disabled.
+        """
+        if not self.scan_assembled_prompt:
+            return PromptScanResult(ok=True)
+        if not prompt:
+            return PromptScanResult(ok=True)
+
+        if self.scanner is not None:
+            try:
+                result = self.scanner(prompt)
+            except Exception as e:  # pragma: no cover - defensive
```

Ruff (0.15.18)

[warning] Do not catch blind exception: `Exception`

(BLE001)